src/MonoTorrent/MonoTorrent.Client/Modes/MetadataMode.cs

```diff
@@ -189,7 +189,15 @@ protected override void HandleLtMetadataMessage (PeerId id, LTMetadata message)
                                 try {
                                     if (!Directory.Exists (Path.GetDirectoryName (savePath)))
                                         Directory.CreateDirectory (Path.GetDirectoryName (savePath));
-                                    File.Delete (savePath);
+
+                                    if(true == File.Exists(savePath))
+                                        File.Delete (savePath);
+                                    else//savePath is a directory
+                                    {
+                                        savePath += "\\" + t.Name + ".torrent";
+                                        using var fs = File.Create(savePath);
+                                    }
```

Comment:
I'd like to do a more holistic refactor so that all string parameters passed to the library are renamed so it's super clear whether they refer to a `directory`or a `file` At the moment it's pretty unclear whether a given string is supposed to be an actual file path or a directory name.  It's also unclear whether something is a full path or relative path.

It shouldn't be this confusing!

I love the direction this change is moving. I think with a little more planning we can perfect this!

Reply:
I don't want to do this larger change *right now* as i think there's value in merging your patch today without trying to fix up the rest of the library. I think we should document this value as being a *directory*, and internally let's always treat it as a directory. Keeping the `if (File.Exists(path)) File.delete(path)` check should be safe as it ensures we can always create a directory in that path to store our metadata in.

Could you submit this section in it's own PR so I can review/test it before merging? I'd love to test these scenarios to see what happens. I think the first scenario will still cause an issue with this modified code.

1 - pass `save\with\sub-dir` as the metadata save path. Do not create any of those directories to begin with.
2 - pass `save` as the metadata safe path. Do not create the `save` directory first